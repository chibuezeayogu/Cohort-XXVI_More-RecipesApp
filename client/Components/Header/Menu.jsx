import React from 'react';
import { withRouter, Link } from 'react-router-dom';

/**
 *
 * @description menu component for unathenticated users
 *
 * @constant
 *
 * @returns {Undefined}
 */
export const Menu = () => (
  <div>
    <nav>
      <div className="nav-wrapper black">
        <a className="brand-logo">
          <img src={'https://res.cloudinary.com/chibuezeayogu/image/upload'+
            '/v1516130489/yy3vdswodkvr3mj5lts1.jpg'}
            className="responsive-img image-radius" 
            alt="" style={{ height: 25, marginRight: 10, marginLeft: 10 }} />
            More Recipes</a>
        <a 
          data-activates="mobile-demo" 
          className="button-collapse">
          <i className="material-icons">menu</i>
        </a>
        <ul className="right hide-on-med-and-down">
          <li>
            <Link to="/">
              <i className="fa fa-home" /> Home
            </Link>
          </li>
          <li>
              <Link to="/signin">
                <i 
                  className="fa fa-sign-in" 
                  aria-hidden="true" />  Login
                </Link>
              </li>
          <li>
            <Link to="/signup">
              <i 
                className="fa fa-user-plus" 
                aria-hidden="true" 
              /> Sign Up
            </Link>
          </li>
        </ul>
        <ul 
          className="side-nav" 
          id="mobile-demo">
          <li>
            <Link to="/">
              <i className="fa fa-home" 
              /> Home
            </Link>
          </li>
          <div className="divider" />
          <li>
            <Link to="/signin">
              <i 
                className="fa fa-sign-in" 
                aria-hidden="true" 
              /> Login
            </Link>
          </li>
          <div className="divider" />
          <li>
            <Link to="/signup">
              <i 
                className="fa fa-user-plus" 
                aria-hidden="true" 
              /> Sign Up
            </Link>
          </li>
          <div className="divider" />
        </ul>
      </div>
    </nav>
  </div>
);
 
export default withRouter(Menu);
