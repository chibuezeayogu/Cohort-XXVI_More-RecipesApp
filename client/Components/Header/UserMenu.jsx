import React, { Component } from 'react';
import PropTypes from 'prop-types';
import jwtDecode from 'jwt-decode';
import { connect } from 'react-redux';
import { SignOut } from '../../action/actionCreators';
import { Link } from 'react-router-dom';
import image from '../../img/logo.jpg';

/**
 * menu component athenticated users
 * @class
 *
 * @extends Component
 *
 */
class UserMenu extends Component {
  /**
   * @description initializes dropdown and button collapse
   *
   * @method
   *
   * @memberOf UserMenu
   *
   * @returns {void}
   *
   */
  componentDidMount() {
    $('.dropdown-button').dropdown({
      inDuration: 300,
      outDuration: 225,
      hover: true,
      belowOrigin: true,
      alignment: 'right',
    });
    $('.button-collapse').sideNav();
  }

  /**
   * @description routes to /if user is logs' out
   *
   * @method
   *
   * @memberOf UserMenu
   *
   * @param {Event} e
   *
   * @returns {void}
   *
   */
  logout(e) {
    e.preventDefault();
    this.props.SignOut();
    localStorage.removeItem('jwtToken');
    this.context.router.history.push('/signin');
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignIn
   *
   * @returns {void}
   *
   */
  render() {
    const token = localStorage.getItem('jwtToken');
    const userData = jwtDecode(token);
    return (
      <div>
        <div>
          <nav>
            <div className="nav-wrapper black">
              <a href="#!" className="brand-logo">
                <img
                  src={image}
                  className="responsive-img image-raduis"
                  alt=""
                  style={{ height: 25 }}
                />More Recipes
              </a>
              <a
                href="#!"
                data-activates="mobile-demo"
                className="button-collapse"
              >
                <i className="material-icons">menu</i>
              </a>
              <ul className="right hide-on-med-and-down">
                <li >
                  <Link to="/recipes">
                    <i className="fa fa-home" /> Recipes
                  </Link>
                </li>
                <li>
                  <Link to="/addrecipe">
                    <i className="fa fa-plus" aria-hidden="true" /> Add Recipe
                  </Link>
                </li>
                <li>
                  <Link to="/search">
                    <i className="fa fa-search" aria-hidden="true" /> Search
                  </Link>
                </li>
                <li>
                  <a
                    href="#!"
                    className="dropdown-button"
                    data-activates="dropdown"
                    data-beloworigin="true"
                  > {userData.user.firstName}
                    <i className="material-icons right">arrow_drop_down</i>
                  </a>
                </li>
              </ul>
              <ul
                className="side-nav"
                id="mobile-demo"
              >
                <li>
                  <Link to="/recipes">
                    <i className="fa fa-home" /> Recipes
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/addrecipe">
                    <i className="fa fa-plus" aria-hidden="true" /> Add Recipe
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/search">
                    <i className="fa fa-search" aria-hidden="true" /> Search
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/user/profile">
                    <i className="fa fa-user" aria-hidden="true" /> Profile
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/user/recipes">
                    <i className="fa fa-cutlery" aria-hidden="true" /> My Recipes
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/user/favourites">
                    <i className="fa fa-heart" aria-hidden="true" /> My Favourites
                  </Link>
                </li>
                <div className="divider" />
                <li>
                  <Link to="/" onClick={e => this.logout(e)}>
                    <i className="fa fa-sign-out" aria-hidden="true" /> Logout
                  </Link>
                </li>
              </ul>
            </div>
          </nav>
        </div>

        <ul id="dropdown" className="dropdown-content black-text">
          <div className="divider" />
          <li className="black-text">
            <Link to="/user/profile" className="black-text">
              <i 
                className="fa fa-user black-text" 
                aria-hidden="true" /> My Profile
            </Link>
          </li>
          <div className="divider" />
          <li className="black-text">
            <Link to="/user/recipes" className="black-text">
              <i className="fa fa-cutlery black-text" 
                aria-hidden="true" /> My Recipes
            </Link>
          </li>
          <div className="divider" />
          <li className="black-text">
            <Link to="/user/favourites" className="black-text">
              <i className="fa fa-heart black-text" 
                aria-hidden="true" /> My Favourites
            </Link>
          </li>
          <div className="divider" />
          <li className="black-text">
            <Link to="/" onClick={e => this.logout(e)} className="black-text">
              <i className="fa fa-sign-out black-text"
                aria-hidden="true" /> Logout
            </Link>
          </li>
        </ul>
      </div>
    );
  }
}

UserMenu.contextTypes = {
  router: PropTypes.object.isRequired
};

UserMenu.propTypes = {
  SignOut: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  userData: state.userData,
});

export default connect(mapStateToProps, { SignOut })(UserMenu);
