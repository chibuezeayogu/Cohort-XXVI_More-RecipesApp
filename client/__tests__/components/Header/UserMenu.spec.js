import React from 'react';
import expect from 'expect';
import { mount, shallow } from 'enzyme';
import { UserMenu } from '../../../Components/Header/UserMenu';

const props = {
  userData: { currentUser: { id: 1, email: 'chibuezeayogu@hotmail.com' } },
  signOut: jest.fn(),
  history: {
    push: jest.fn()
  },
};

describe('UserMenu Component', () => {
  it('render the user menu without crashing', () => {
    const wrapper = shallow(<UserMenu {...props} />);
    wrapper.instance().componentDidMount();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<UserMenu />', () => {
  it('shoud call logout function', () => {
    const wrapper = shallow(<UserMenu {...props} />);
    const logoutSpy = jest.spyOn(wrapper.instance(), 'logout');
    wrapper.instance().logout();
    expect(logoutSpy).toHaveBeenCalled();
    expect(wrapper.instance().props.signOut).toHaveBeenCalled();
    expect(wrapper.instance().props.history.push).toHaveBeenCalled();
  });
});
