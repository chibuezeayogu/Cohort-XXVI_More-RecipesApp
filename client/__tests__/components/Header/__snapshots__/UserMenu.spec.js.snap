// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMenu /> render the user menu without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div>
    <div>
      <nav>
        <div
          className="nav-wrapper black"
        >
          <a
            className="brand-logo"
            href="#!"
          >
            <img
              alt=""
              className="responsive-img image-radius"
              src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
              style={
                Object {
                  "height": 25,
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            />
            More Recipes
          </a>
          <a
            className="button-collapse"
            data-activates="mobile-demo"
            href="#!"
          >
            <i
              className="material-icons"
            >
              menu
            </i>
          </a>
          <ul
            className="right hide-on-med-and-down"
          >
            <li>
              <Link
                replace={false}
                to="/recipes"
              >
                <i
                  className="fa fa-home"
                />
                 Recipes
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/addrecipe"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-plus"
                />
                 Add Recipe
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/search"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-search"
                />
                 Search
              </Link>
            </li>
            <li>
              <a
                className="dropdown-button"
                data-activates="dropdown"
                data-beloworigin="true"
                href="#!"
                id="displayEmail"
              >
                 
                chibuezeayogu@hotmail.com
                <i
                  className="material-icons right"
                >
                  arrow_drop_down
                </i>
              </a>
            </li>
          </ul>
          <ul
            className="side-nav"
            id="mobile-demo"
          >
            <li>
              <Link
                replace={false}
                to="/recipes"
              >
                <i
                  className="fa fa-home"
                />
                 Recipes
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                replace={false}
                to="/addrecipe"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-plus"
                />
                 Add Recipe
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                replace={false}
                to="/search"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-search"
                />
                 Search
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                replace={false}
                to="/user/profile"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user"
                />
                 Profile
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                replace={false}
                to="/user/recipes"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-cutlery"
                />
                 My Recipes
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                replace={false}
                to="/user/favourites"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-heart"
                />
                 My Favourites
              </Link>
            </li>
            <div
              className="divider"
            />
            <li>
              <Link
                onClick={[Function]}
                replace={false}
                to="/signin"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-sign-out"
                />
                 Logout
              </Link>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <ul
      className="dropdown-content black-text"
      id="dropdown"
    >
      <div
        className="divider"
      />
      <li
        className="black-text"
        id="profile"
      >
        <Link
          className="black-text"
          replace={false}
          to="/user/profile"
        >
          <i
            aria-hidden="true"
            className="fa fa-user black-text"
          />
           My Profile
        </Link>
      </li>
      <div
        className="divider"
      />
      <li
        className="black-text"
        id="myRecipes"
      >
        <Link
          className="black-text"
          replace={false}
          to="/user/recipes"
        >
          <i
            aria-hidden="true"
            className="fa fa-cutlery black-text"
          />
           My Recipes
        </Link>
      </li>
      <div
        className="divider"
      />
      <li
        className="black-text"
        id="myFavourite"
      >
        <Link
          className="black-text"
          replace={false}
          to="/user/favourites"
        >
          <i
            aria-hidden="true"
            className="fa fa-heart black-text"
          />
           My Favourites
        </Link>
      </li>
      <div
        className="divider"
      />
      <li
        className="black-text"
        id="logout"
      >
        <Link
          className="black-text"
          onClick={[Function]}
          replace={false}
          to="/signin"
        >
          <i
            aria-hidden="true"
            className="fa fa-sign-out black-text"
          />
           Logout
        </Link>
      </li>
    </ul>
  </div>,
  "nodes": Array [
    <div>
      <div>
        <nav>
          <div
            className="nav-wrapper black"
          >
            <a
              className="brand-logo"
              href="#!"
            >
              <img
                alt=""
                className="responsive-img image-radius"
                src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
                style={
                  Object {
                    "height": 25,
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              />
              More Recipes
            </a>
            <a
              className="button-collapse"
              data-activates="mobile-demo"
              href="#!"
            >
              <i
                className="material-icons"
              >
                menu
              </i>
            </a>
            <ul
              className="right hide-on-med-and-down"
            >
              <li>
                <Link
                  replace={false}
                  to="/recipes"
                >
                  <i
                    className="fa fa-home"
                  />
                   Recipes
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/addrecipe"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-plus"
                  />
                   Add Recipe
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/search"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-search"
                  />
                   Search
                </Link>
              </li>
              <li>
                <a
                  className="dropdown-button"
                  data-activates="dropdown"
                  data-beloworigin="true"
                  href="#!"
                  id="displayEmail"
                >
                   
                  chibuezeayogu@hotmail.com
                  <i
                    className="material-icons right"
                  >
                    arrow_drop_down
                  </i>
                </a>
              </li>
            </ul>
            <ul
              className="side-nav"
              id="mobile-demo"
            >
              <li>
                <Link
                  replace={false}
                  to="/recipes"
                >
                  <i
                    className="fa fa-home"
                  />
                   Recipes
                </Link>
              </li>
              <div
                className="divider"
              />
              <li>
                <Link
                  replace={false}
                  to="/addrecipe"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-plus"
                  />
                   Add Recipe
                </Link>
              </li>
              <div
                className="divider"
              />
              <li>
                <Link
                  replace={false}
                  to="/search"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-search"
                  />
                   Search
                </Link>
              </li>
              <div
                className="divider"
              />
              <li>
                <Link
                  replace={false}
                  to="/user/profile"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user"
                  />
                   Profile
                </Link>
              </li>
              <div
                className="divider"
              />
              <li>
                <Link
                  replace={false}
                  to="/user/recipes"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-cutlery"
                  />
                   My Recipes
                </Link>
              </li>
              <div
                className="divider"
              />
              <li>
                <Link
                  replace={false}
                  to="/user/favourites"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-heart"
                  />
                   My Favourites
                </Link>
              </li>
              <div
                className="divider"
              />
              <li>
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/signin"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-sign-out"
                  />
                   Logout
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </div>
      <ul
        className="dropdown-content black-text"
        id="dropdown"
      >
        <div
          className="divider"
        />
        <li
          className="black-text"
          id="profile"
        >
          <Link
            className="black-text"
            replace={false}
            to="/user/profile"
          >
            <i
              aria-hidden="true"
              className="fa fa-user black-text"
            />
             My Profile
          </Link>
        </li>
        <div
          className="divider"
        />
        <li
          className="black-text"
          id="myRecipes"
        >
          <Link
            className="black-text"
            replace={false}
            to="/user/recipes"
          >
            <i
              aria-hidden="true"
              className="fa fa-cutlery black-text"
            />
             My Recipes
          </Link>
        </li>
        <div
          className="divider"
        />
        <li
          className="black-text"
          id="myFavourite"
        >
          <Link
            className="black-text"
            replace={false}
            to="/user/favourites"
          >
            <i
              aria-hidden="true"
              className="fa fa-heart black-text"
            />
             My Favourites
          </Link>
        </li>
        <div
          className="divider"
        />
        <li
          className="black-text"
          id="logout"
        >
          <Link
            className="black-text"
            onClick={[Function]}
            replace={false}
            to="/signin"
          >
            <i
              aria-hidden="true"
              className="fa fa-sign-out black-text"
            />
             Logout
          </Link>
        </li>
      </ul>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserMenu
        history={
          Object {
            "push": [MockFunction],
          }
        }
        signOut={[MockFunction]}
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "logout": [Function],
        "props": Object {
          "history": Object {
            "push": [MockFunction],
          },
          "signOut": [MockFunction],
          "userData": Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div>
            <nav>
              <div
                className="nav-wrapper black"
              >
                <a
                  className="brand-logo"
                  href="#!"
                >
                  <img
                    alt=""
                    className="responsive-img image-radius"
                    src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
                    style={
                      Object {
                        "height": 25,
                        "marginLeft": 10,
                        "marginRight": 10,
                      }
                    }
                  />
                  More Recipes
                </a>
                <a
                  className="button-collapse"
                  data-activates="mobile-demo"
                  href="#!"
                >
                  <i
                    className="material-icons"
                  >
                    menu
                  </i>
                </a>
                <ul
                  className="right hide-on-med-and-down"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/recipes"
                    >
                      <i
                        className="fa fa-home"
                      />
                       Recipes
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/addrecipe"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-plus"
                      />
                       Add Recipe
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/search"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-search"
                      />
                       Search
                    </Link>
                  </li>
                  <li>
                    <a
                      className="dropdown-button"
                      data-activates="dropdown"
                      data-beloworigin="true"
                      href="#!"
                      id="displayEmail"
                    >
                       
                      chibuezeayogu@hotmail.com
                      <i
                        className="material-icons right"
                      >
                        arrow_drop_down
                      </i>
                    </a>
                  </li>
                </ul>
                <ul
                  className="side-nav"
                  id="mobile-demo"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/recipes"
                    >
                      <i
                        className="fa fa-home"
                      />
                       Recipes
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/addrecipe"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-plus"
                      />
                       Add Recipe
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/search"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-search"
                      />
                       Search
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/user/profile"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user"
                      />
                       Profile
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/user/recipes"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-cutlery"
                      />
                       My Recipes
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/user/favourites"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-heart"
                      />
                       My Favourites
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      onClick={[Function]}
                      replace={false}
                      to="/signin"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-sign-out"
                      />
                       Logout
                    </Link>
                  </li>
                </ul>
              </div>
            </nav>
          </div>
          <ul
            className="dropdown-content black-text"
            id="dropdown"
          >
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="profile"
            >
              <Link
                className="black-text"
                replace={false}
                to="/user/profile"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user black-text"
                />
                 My Profile
              </Link>
            </li>
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="myRecipes"
            >
              <Link
                className="black-text"
                replace={false}
                to="/user/recipes"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-cutlery black-text"
                />
                 My Recipes
              </Link>
            </li>
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="myFavourite"
            >
              <Link
                className="black-text"
                replace={false}
                to="/user/favourites"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-heart black-text"
                />
                 My Favourites
              </Link>
            </li>
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="logout"
            >
              <Link
                className="black-text"
                onClick={[Function]}
                replace={false}
                to="/signin"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-sign-out black-text"
                />
                 Logout
              </Link>
            </li>
          </ul>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div>
            <nav>
              <div
                className="nav-wrapper black"
              >
                <a
                  className="brand-logo"
                  href="#!"
                >
                  <img
                    alt=""
                    className="responsive-img image-radius"
                    src="https://res.cloudinary.com/chibuezeayogu/image/upload/v1516130489/yy3vdswodkvr3mj5lts1.jpg"
                    style={
                      Object {
                        "height": 25,
                        "marginLeft": 10,
                        "marginRight": 10,
                      }
                    }
                  />
                  More Recipes
                </a>
                <a
                  className="button-collapse"
                  data-activates="mobile-demo"
                  href="#!"
                >
                  <i
                    className="material-icons"
                  >
                    menu
                  </i>
                </a>
                <ul
                  className="right hide-on-med-and-down"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/recipes"
                    >
                      <i
                        className="fa fa-home"
                      />
                       Recipes
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/addrecipe"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-plus"
                      />
                       Add Recipe
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/search"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-search"
                      />
                       Search
                    </Link>
                  </li>
                  <li>
                    <a
                      className="dropdown-button"
                      data-activates="dropdown"
                      data-beloworigin="true"
                      href="#!"
                      id="displayEmail"
                    >
                       
                      chibuezeayogu@hotmail.com
                      <i
                        className="material-icons right"
                      >
                        arrow_drop_down
                      </i>
                    </a>
                  </li>
                </ul>
                <ul
                  className="side-nav"
                  id="mobile-demo"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/recipes"
                    >
                      <i
                        className="fa fa-home"
                      />
                       Recipes
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/addrecipe"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-plus"
                      />
                       Add Recipe
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/search"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-search"
                      />
                       Search
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/user/profile"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user"
                      />
                       Profile
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/user/recipes"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-cutlery"
                      />
                       My Recipes
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      replace={false}
                      to="/user/favourites"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-heart"
                      />
                       My Favourites
                    </Link>
                  </li>
                  <div
                    className="divider"
                  />
                  <li>
                    <Link
                      onClick={[Function]}
                      replace={false}
                      to="/signin"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-sign-out"
                      />
                       Logout
                    </Link>
                  </li>
                </ul>
              </div>
            </nav>
          </div>
          <ul
            className="dropdown-content black-text"
            id="dropdown"
          >
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="profile"
            >
              <Link
                className="black-text"
                replace={false}
                to="/user/profile"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user black-text"
                />
                 My Profile
              </Link>
            </li>
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="myRecipes"
            >
              <Link
                className="black-text"
                replace={false}
                to="/user/recipes"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-cutlery black-text"
                />
                 My Recipes
              </Link>
            </li>
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="myFavourite"
            >
              <Link
                className="black-text"
                replace={false}
                to="/user/favourites"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-heart black-text"
                />
                 My Favourites
              </Link>
            </li>
            <div
              className="divider"
            />
            <li
              className="black-text"
              id="logout"
            >
              <Link
                className="black-text"
                onClick={[Function]}
                replace={false}
                to="/signin"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-sign-out black-text"
                />
                 Logout
              </Link>
            </li>
          </ul>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserMenu
    history={
      Object {
        "push": [MockFunction],
      }
    }
    signOut={[MockFunction]}
    userData={
      Object {
        "currentUser": Object {
          "email": "chibuezeayogu@hotmail.com",
          "id": 1,
        },
      }
    }
  />,
}
`;
